### PR TITLE
WIP: Fixes to make application working

### PR DESCRIPTION
**Pull request by MikeAndrianov**

### CODE REVIEW

File: lib/bepaid.ex [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 
-  def start_link do
+
+  def start(_type, _args) do
     import Supervisor.Spec, warn: false
 
     children = []
 
     opts = [strategy: :one_for_one, name: Bepaid.Supervisor]
-    {:ok, _pid} =  Supervisor.start_link(children, opts)
```
Comment (paveltyk) [inline lib/bepaid.ex:15]: @MikeAndrianov why we do not check if the app has started successfully?

File: lib/bepaid/gateway.ex [hunk 2]
```diff
@@ -50,7 +58,6 @@ defmodule Bepaid.Gateway do
   defp headers_for(:get), do: [auth_header(), {"Accept", "application/json"}]
   defp auth_header, do: {"Authorization", "Basic " <> Base.encode64("#{get_env_var(@shop_id)}:#{get_env_var(@key_secret)}")}
 
-  defp get_env_var({:system, env_var}), do: System.get_env(env_var)
```
Comment (paveltyk) [inline lib/bepaid/gateway.ex]: @MikeAndrianov I've left a comment in another repo about this.

File: lib/bepaid/gateway_behaviour.ex [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  @moduledoc """
+  Interface for Gateway. Could be used for defining your own module for testing
+  """
+
+  alias Bepaid.Payment
+
+  @type http_response :: {:ok, map} | {:error, any}
+
+  @doc """
+  """
```
Comment (paveltyk) [inline lib/bepaid/gateway_behaviour.ex]: @MikeAndrianov is this a good place to provide docs? If so, I will input description here. Please let me know.

File: README.md [hunk 4]
```diff
@@ -15,6 +15,21 @@ def deps do
 end
 ```
 
-Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
-and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
-be found at [https://hexdocs.pm/bepaid](https://hexdocs.pm/bepaid).
+## Configuration
```
Comment (paveltyk) [inline README.md]: @MikeAndrianov once you figure out how to run tests in parent app, will need to provide instruction how to setup test env with `bepaid_ex` library.

**Review by paveltyk — CHANGES_REQUESTED** [review]
Look great. Thank you for putting so much effort into this.

File: README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
-# Bepaid Elixir Library
```
Comment (paveltyk) [inline README.md:1]: @MikeAndrianov lets use "Bepaid" as a library name. Though gateway service name is "bePaid", I prefer to keep it simple.

File: lib/bepaid.ex (hunk 1 above)
Comment (MikeAndrianov) [reply lib/bepaid.ex:15]: Don't remember exactly :| 
I will put it back

File: lib/bepaid/gateway_behaviour.ex (hunk 3 above)
Comment (MikeAndrianov) [reply lib/bepaid/gateway_behaviour.ex]: As I can see in ecto repo, they cover behaviours [with docs](https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/type.ex), but maybe we could get rid of them for now.

**Pull request merged**